### PR TITLE
Adding Model Getters for SOS, SD, and NL Constraint Counts

### PR DESCRIPTION
**Pull request by ccoffrin**
A hack at adding this without understanding the best way to do it.  Suggestions welcome.

### CODE REVIEW

**Comment by mlubin** [comment]
Seems reasonable to me. Could you update the corresponding documentation in ``doc/refmodel.rst``?

File: src/JuMP.jl [hunk 1]
```diff
@@ -188,8 +188,13 @@ end
 MathProgBase.numvar(m::Model) = m.numCols
 MathProgBase.numlinconstr(m::Model) = length(m.linconstr)
 MathProgBase.numquadconstr(m::Model) = length(m.quadconstr)
+# Note SOC constraints are captured in numquadconstr
```
Comment (mlubin) [inline src/JuMP.jl]: SOC constraints are only captured in ``numquadconstr`` if they're specified as quadratic constraints. If they're specified as SOC constraints using the ``norm`` syntax then they will not be counted there.

Comment (ccoffrin) [inline src/JuMP.jl]: Ok, I'll fix that too.

**Comment by ccoffrin** [comment]
I think both comments are addressed now.

**Comment by mlubin** [comment]
Thanks!

**Pull request merged**